src/pages/_document.page.tsx

```diff
@@ -25,9 +25,11 @@ class MyDocument extends Document<DocumentProps> {
         <Head>
           <Decorator.Styles />
         </Head>
-        <body className="h-screen">
+        <body className="h-screen flex flex-col">
           <Decorator.Header />
-          <Main />
+          <div className="flex-grow">
```

Comment:
må ikke flex-grow ha en verdi?

ser ut som tailwind sine klasser heter "grow" eller "grow-0".

Reply:
Og er flex-grow riktig her / er det egentlig h-screen eller h-full du er på jakt etter her, dersom det skal ha full height?

Reply:
Med h-screen dytter vi footer ut av viewport + den adder på tilsvarende space fra head i main content. Vil vi ikke heller vise footer så lenge main content ikke fyller hele viewporten? Maincontent vil også fylle rest space som gjort i denne PRen tror jeg

Reply:
Ser ut som h-full også funker 👍

Reply:
Jeg testa på fredag og da fikk jeg ikke hverken h-full eller grow til å funke, men nå fungerer det.
Vet ikke hva jeg gjorde annerledes, kanskje det hadde noe med kombinasjonen med `flex flex-col` å gjøre. 🤷‍♀️  Jeg kan endre til `h-full` hvis det er det som er best.

til spørsmål om flex-grow må ha verdi: Det ser ut som tailwind oversetter fra flex-grow til `flex-grow: 1`.

Reply:
Side note: Hvorfor er det egentlig bedre å bruke `h-full` enn `grow`? Det er jo flere elementer på siden, og det er bare main content som skal bli større. Basert på beskrivelsene ville jeg tenkt at det er grow som er riktig å bruke. 

- Use grow to allow a flex item to grow to fill any available space
- Use h-full to set an element’s height to 100% of its parent, as long as the parent has a defined height.

Reply:
Mystisk! Ser det ikke i tailwind-doc: https://tailwindcss.com/docs/flex-grow

Så lenge det funker er det jo topp.

Jeg er litt usikker på hvordan dere ønsker å ha layout, om dere vil bruke 100% (skjerm)-høyde alltid, eller om dere vil det skal ha litt ulik høyde basert på innholdet på siden.

Jeg har tidligere brukt mye flex-1 for å fikse det, ref https://tailwindcss.com/docs/flex

Reply:
Jeg stussa bare litt på hvorfor className="flex-grow" fungerte da jeg ikke så det i tailwind docs :D

Reply:
Flex er en shorthand for flex-grow: https://developer.mozilla.org/en-US/docs/Web/CSS/flex-grow#description

I vårt tilfelle tror jeg ikke det har noe å si om vi bruker h-full eller flex men i noen tilfeller der man feks ønsker at flere komponenter sammen skal fylle så bruker man flex på disse så vil de automatisk fylles riktig, det føles mer eksplisitt ut. Derfor pleier jeg bare å bruke flex generelt når jeg kan. Usikker på hvordan det blir med flere komponenter med h-full, kanskje det er det samme :p

Reply:
Ja, men den fyller samtidig ut flex-shrink og flex-basis :)

Nei kan hende det blir samme :D

Reply:
Tror det står at det er vanlig å bruke det sammen med de andre reglene:

> flex-grow is used **alongside** the other flex properties [flex-shrink](https://developer.mozilla.org/en-US/docs/Web/CSS/flex-shrink) and [flex-basis](https://developer.mozilla.org/en-US/docs/Web/CSS/flex-basis), and normally defined using the [flex](https://developer.mozilla.org/en-US/docs/Web/CSS/flex) shorthand to ensure all values are set.

Reply:
Sant, jeg ser det på som en forenkling av det man prøver å oppnå med shorthanden, det blir ikke ofte feil å bruke flex hvis man tenker å bruke flex-grow

Reply:
Det blir `grow`. Fungerte ikke så godt med veldig zoomet inn (200%) + smal viewport i kombinasjon med `h-full.` Da ble det overlapp mellom skjema og footer.